_posts/2019-12-05-testowanie-komponentow-z-inputami-i-outputami.md

```diff
@@ -0,0 +1,198 @@
+---
+layout:    post
+title:     Testowanie frontendu - Cz. 3. Testowanie komponentów angularowych z inputami i outputami
+published: true
+date:      2019-12-05 08:00:00 +0100
+author:    pgrobelny
+tags:
+    - Jasmine
+    - Angular
+    - unit test
+    - komponent
+---
+## Wprowadzenie
+Czas na kolejną dawkę informacji dotyczących testowania przy użyciu Jasmine. Po przeczytaniu wcześniejszych wpisów ([Cz. 1]({% post_url 2019-11-06-testowanie-frontendu-wprowadzenie-do-jasmine %}) i [Cz. 2]({% post_url 2019-11-20-testowanie-komponentow-i-serwisow %})) pora skupić się na testowaniu komponentów angularowych, a w szczególności ich wejść i wyjść. Przykłady oprzemy o aplikację, która będzie składała się kilku drobnych elementów wzajemnie się ze sobą komunikujących.
+
+Schemat naszej przykładowej aplikacji wygląda następująco:
+
+![diagram komponentów w aplikacji](/assets/img/posts/2019-12-05-testowanie-komponentow-z-inputami-i-outputami/diagram.jpg)
+
+Jest to prosty program mający na celu dodawanie i zapisywanie wyników. Został rozpisany na komponenty tak, aby można było spokojnie przetestować wejścia i wyjścia w różnych wariantach.
```

Comment:
Pierwsze zdanie dobrze by było przeredagować, bo obecnie brzmi, jakby to diagram dodawał i zapisywał inputy, a nie aplikacja :)

Reply:
Uwaga nie na miejscu, bo przeczytałem "program" jako "diagram" o.O Musiało mi się zlać z nazwą pliku z odnośnika wyżej. Cofam :D